src/main/java/frc/robot/subsystems/arm/ArmIOReal.java

```diff
@@ -0,0 +1,324 @@
+        shoulderMainMotorConfig.Feedback.SensorToMechanismRatio = SENSOR_TO_MECHANISM_RATIO;
+        shoulderMainMotorConfig.Feedback.RotorToSensorRatio = ROTOR_TO_SENSOR_RATIO;
+        shoulderMainMotor.getConfigurator().apply(shoulderMainMotorConfig);
+
+        shoulderMainMotor.setNeutralMode(NeutralModeValue.Brake);
+        shoulderMainMotor.setInverted(MAIN_CLOCKWISE);
+
+        elbowMainMotorConfig.Voltage.PeakForwardVoltage = ArmConstants.VOLT_COMP_SATURATION;
+        elbowMainMotorConfig.Voltage.PeakReverseVoltage = -ArmConstants.VOLT_COMP_SATURATION;
+        elbowMainMotorConfig.Slot1 = ELBOW_PID;
+        elbowMainMotorConfig.CurrentLimits.SupplyCurrentLimit = ArmConstants.CURRENT_LIMIT;
+        elbowMainMotorConfig.CurrentLimits.StatorCurrentLimit = ArmConstants.CURRENT_LIMIT;
+        elbowMainMotorConfig.Feedback.SensorToMechanismRatio = SENSOR_TO_MECHANISM_RATIO;
+        elbowMainMotorConfig.Feedback.RotorToSensorRatio = ROTOR_TO_SENSOR_RATIO;
+        elbowMainMotor.getConfigurator().apply(elbowMainMotorConfig);
+
+        elbowMainMotor.setNeutralMode(NeutralModeValue.Brake);
+        elbowMainMotor.setInverted(MAIN_CLOCKWISE);
+
+    }
+
+    /**
+     * Get the instance of the arm subsystem.
+     *
+     * @return Arm instance
+     */
+    public static ArmIOReal getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new ArmIOReal();
+        }
+        return INSTANCE;
+    }
+
+
+    /**
+     * Sets the power of the shoulder motors.
+     *
+     * @param power desired power. [-1,1]
+     */
+    public void setShoulderJointPower(double power) {
+        shoulderMainMotor.setControl(new DutyCycleOut(power));
+    }
+
+    public double getShoulderJointPower() {
+        return shoulderMainMotor.get();
+    }
+
+    public double getElbowJointPower() {
+        return elbowMainMotor.get();
+    }
+
+    /**
+     * Sets the power of the elbow motors.
+     *
+     * @param power desired power. [-1,1]
+     */
+    public void setElbowJointPower(double power) {
+        elbowMainMotor.setControl(new DutyCycleOut(power));
+    }
+
+
+    /**
+     * Sets the angle of the shoulder joint.
+     *
+     * @param angle desired angle. [degrees]
+     */
+    public void setShoulderJointAngle(Rotation2d angle) {
+        setShoulderJointAngle(angle.getRadians(), SHOULDER_FEED_FORWARD_MULTIPLIER, SHOULDER_VELOCITY);
+    }
+
+
+    private void setShoulderJointAngle(double angle, double ffMultiplier, int velocity) {
+        angle = AngleUtil.normalize(angle);
+        Rotation2d error = new Rotation2d(angle).minus(new Rotation2d(currentShoulderAngle));
+        if (shoulderEncoder.isConnected()) {
+            motorControlRequest = new PositionVoltage(angle);
+            shoulderMainMotor.setControl(new PositionDutyCycle(angle, velocity, true,
+                    shoulderFeedforward * ffMultiplier, 0, true)
+                    .withEnableFOC(true));
```

Comment:
Same thing as the duty cycle request:
```suggestion
            shoulderMainMotor.setControl(positionDutyCycleShoulderRequest
                    .withPosition(angle)
                    .withVelocity(velocity)
                    .withFeedForward(shoulderFeedforward * ffMultiplier)
                    .withEnableFOC(true));
```
Also, note the difference between the `PositionDutyCycle` which takes a feedfoward in output (-1 to 1), and the `PositionVoltage` which takes a feedforward in voltage.